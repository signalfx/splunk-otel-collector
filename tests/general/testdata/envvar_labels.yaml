receivers:
  hostmetrics:
    collection_interval: 2s
    scrapers:
      memory:
processors:
  filter:
    metrics:
      include:
        match_type: strict
        metric_names:
          - system.memory.usage
  metricstransform:
    transforms:
      - action: update
        include: .*
        match_type: regexp
        operations:
          - action: add_label
            new_label: single-dollar-no-curly-braces
            new_value: $AN_ENVVAR-$nonenvvar-$-suffix $
          - action: add_label
            new_label: single-dollar-curly-braces
            new_value: ${AN_ENVVAR}-$nonenvvar-$-suffix $
          - action: add_label
            new_label: double-dollar-no-curly-braces
            new_value: $$AN_ENVVAR-$$nonenvvar-$$-suffix $$
          - action: add_label
            new_label: double-dollar-curly-braces
            new_value: $${AN_ENVVAR}-$$nonenvvar-$$-suffix $$
          - action: add_label
            new_label: triple-dollar-no-curly-braces
            new_value: $$$AN_ENVVAR-$$$nonenvvar-$$$-suffix $$$
          - action: add_label
            new_label: triple-dollar-curly-braces
            new_value: $$${AN_ENVVAR}-$$$nonenvvar-$$$-suffix $$$
          - action: add_label
            new_label: quadruple-dollar-no-curly-braces
            new_value: $$$$AN_ENVVAR-$$$$nonenvvar-$$$$-suffix $$$$
          - action: add_label
            new_label: quadruple-dollar-curly-braces
            new_value: $$$${AN_ENVVAR}-$$$$nonenvvar-$$$$-suffix $$$$
          - action: add_label
            new_label: quintuple-dollar-no-curly-braces
            new_value: $$$$$AN_ENVVAR-$$$$$nonenvvar-$$$$$-suffix $$$$$
          - action: add_label
            new_label: quintuple-dollar-curly-braces
            new_value: $$$$${AN_ENVVAR}-$$$$$nonenvvar-$$$$$-suffix $$$$$
          - action: add_label
            new_label: sextuple-dollar-no-curly-braces
            new_value: $$$$$$AN_ENVVAR-$$$$$$nonenvvar-$$$$$$-suffix $$$$$$
          - action: add_label
            new_label: sextuple-dollar-curly-braces
            new_value: $$$$$${AN_ENVVAR}-$$$$$$nonenvvar-$$$$$$-suffix $$$$$$
          - action: add_label
            new_label: septuple-dollar-no-curly-braces
            new_value: $$$$$$$AN_ENVVAR-$$$$$$$nonenvvar-$$$$$$$-suffix $$$$$$$
          - action: add_label
            new_label: septuple-dollar-curly-braces
            new_value: $$$$$$${AN_ENVVAR}-$$$$$$$nonenvvar-$$$$$$$-suffix $$$$$$$
          - action: add_label
            new_label: octuple-dollar-no-curly-braces
            new_value: $$$$$$$$AN_ENVVAR-$$$$$$$$nonenvvar-$$$$$$$$-suffix $$$$$$$$
          - action: add_label
            new_label: octuple-dollar-curly-braces
            new_value: $$$$$$$${AN_ENVVAR}-$$$$$$$$nonenvvar-$$$$$$$$-suffix $$$$$$$$
          - action: add_label
            new_label: nonuple-dollar-no-curly-braces
            new_value: $$$$$$$$$AN_ENVVAR-$$$$$$$$$nonenvvar-$$$$$$$$$-suffix $$$$$$$$$
          - action: add_label
            new_label: nonuple-dollar-curly-braces
            new_value: $$$$$$$$${AN_ENVVAR}-$$$$$$$$$nonenvvar-$$$$$$$$$-suffix $$$$$$$$$
          - action: add_label
            new_label: decuple-dollar-no-curly-braces
            new_value: $$$$$$$$$$AN_ENVVAR-$$$$$$$$$$nonenvvar-$$$$$$$$$$-suffix $$$$$$$$$$
          - action: add_label
            new_label: decuple-dollar-curly-braces
            new_value: $$$$$$$$$${AN_ENVVAR}-$$$$$$$$$$nonenvvar-$$$$$$$$$$-suffix $$$$$$$$$$

exporters:
  otlp:
    endpoint: "${OTLP_ENDPOINT}"
    insecure: true
service:
  pipelines:
    metrics:
      receivers:
        - hostmetrics
      processors:
        - filter
        - metricstransform
      exporters:
        - otlp
