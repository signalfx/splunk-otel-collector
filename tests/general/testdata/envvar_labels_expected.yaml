resourceMetrics:
  - resource: {}
    schemaUrl: https://opentelemetry.io/schemas/1.9.0
    scopeMetrics:
      - metrics:
          - description: Bytes of memory in use.
            name: system.memory.usage
            sum:
              aggregationTemporality: 2
              dataPoints:
                - asInt: "226607104"
                  attributes:
                    - key: decuple-dollar-curly-braces
                      value:
                        stringValue: $$$$${AN_ENVVAR}-$$$$$-suffix $$$$$
                    - key: double-dollar-curly-braces
                      value:
                        stringValue: ${AN_ENVVAR}-$-suffix $
                    - key: nonuple-dollar-curly-braces
                      value:
                        stringValue: $$$$an-envvar-value-$$$$$-suffix $$$$$
                    - key: octuple-dollar-curly-braces
                      value:
                        stringValue: $$$${AN_ENVVAR}-$$$$-suffix $$$$
                    - key: quadruple-dollar-curly-braces
                      value:
                        stringValue: $${AN_ENVVAR}-$$-suffix $$
                    - key: quintuple-dollar-curly-braces
                      value:
                        stringValue: $$an-envvar-value-$$$-suffix $$$
                    - key: septuple-dollar-curly-braces
                      value:
                        stringValue: $$$an-envvar-value-$$$$-suffix $$$$
                    - key: sextuple-dollar-curly-braces
                      value:
                        stringValue: $$${AN_ENVVAR}-$$$-suffix $$$
                    - key: single-dollar-curly-braces
                      value:
                        stringValue: an-envvar-value-$-suffix $
                    - key: state
                      value:
                        stringValue: buffered
                    - key: triple-dollar-curly-braces
                      value:
                        stringValue: $an-envvar-value-$$-suffix $$
                  startTimeUnixNano: "1000000"
                  timeUnixNano: "2000000"
                - asInt: "4355420160"
                  attributes:
                    - key: decuple-dollar-curly-braces
                      value:
                        stringValue: $$$$${AN_ENVVAR}-$$$$$-suffix $$$$$
                    - key: double-dollar-curly-braces
                      value:
                        stringValue: ${AN_ENVVAR}-$-suffix $
                    - key: nonuple-dollar-curly-braces
                      value:
                        stringValue: $$$$an-envvar-value-$$$$$-suffix $$$$$
                    - key: octuple-dollar-curly-braces
                      value:
                        stringValue: $$$${AN_ENVVAR}-$$$$-suffix $$$$
                    - key: quadruple-dollar-curly-braces
                      value:
                        stringValue: $${AN_ENVVAR}-$$-suffix $$
                    - key: quintuple-dollar-curly-braces
                      value:
                        stringValue: $$an-envvar-value-$$$-suffix $$$
                    - key: septuple-dollar-curly-braces
                      value:
                        stringValue: $$$an-envvar-value-$$$$-suffix $$$$
                    - key: sextuple-dollar-curly-braces
                      value:
                        stringValue: $$${AN_ENVVAR}-$$$-suffix $$$
                    - key: single-dollar-curly-braces
                      value:
                        stringValue: an-envvar-value-$-suffix $
                    - key: state
                      value:
                        stringValue: cached
                    - key: triple-dollar-curly-braces
                      value:
                        stringValue: $an-envvar-value-$$-suffix $$
                  startTimeUnixNano: "1000000"
                  timeUnixNano: "2000000"
                - asInt: "3154911232"
                  attributes:
                    - key: decuple-dollar-curly-braces
                      value:
                        stringValue: $$$$${AN_ENVVAR}-$$$$$-suffix $$$$$
                    - key: double-dollar-curly-braces
                      value:
                        stringValue: ${AN_ENVVAR}-$-suffix $
                    - key: nonuple-dollar-curly-braces
                      value:
                        stringValue: $$$$an-envvar-value-$$$$$-suffix $$$$$
                    - key: octuple-dollar-curly-braces
                      value:
                        stringValue: $$$${AN_ENVVAR}-$$$$-suffix $$$$
                    - key: quadruple-dollar-curly-braces
                      value:
                        stringValue: $${AN_ENVVAR}-$$-suffix $$
                    - key: quintuple-dollar-curly-braces
                      value:
                        stringValue: $$an-envvar-value-$$$-suffix $$$
                    - key: septuple-dollar-curly-braces
                      value:
                        stringValue: $$$an-envvar-value-$$$$-suffix $$$$
                    - key: sextuple-dollar-curly-braces
                      value:
                        stringValue: $$${AN_ENVVAR}-$$$-suffix $$$
                    - key: single-dollar-curly-braces
                      value:
                        stringValue: an-envvar-value-$-suffix $
                    - key: state
                      value:
                        stringValue: free
                    - key: triple-dollar-curly-braces
                      value:
                        stringValue: $an-envvar-value-$$-suffix $$
                  startTimeUnixNano: "1000000"
                  timeUnixNano: "2000000"
                - asInt: "251011072"
                  attributes:
                    - key: decuple-dollar-curly-braces
                      value:
                        stringValue: $$$$${AN_ENVVAR}-$$$$$-suffix $$$$$
                    - key: double-dollar-curly-braces
                      value:
                        stringValue: ${AN_ENVVAR}-$-suffix $
                    - key: nonuple-dollar-curly-braces
                      value:
                        stringValue: $$$$an-envvar-value-$$$$$-suffix $$$$$
                    - key: octuple-dollar-curly-braces
                      value:
                        stringValue: $$$${AN_ENVVAR}-$$$$-suffix $$$$
                    - key: quadruple-dollar-curly-braces
                      value:
                        stringValue: $${AN_ENVVAR}-$$-suffix $$
                    - key: quintuple-dollar-curly-braces
                      value:
                        stringValue: $$an-envvar-value-$$$-suffix $$$
                    - key: septuple-dollar-curly-braces
                      value:
                        stringValue: $$$an-envvar-value-$$$$-suffix $$$$
                    - key: sextuple-dollar-curly-braces
                      value:
                        stringValue: $$${AN_ENVVAR}-$$$-suffix $$$
                    - key: single-dollar-curly-braces
                      value:
                        stringValue: an-envvar-value-$-suffix $
                    - key: state
                      value:
                        stringValue: slab_reclaimable
                    - key: triple-dollar-curly-braces
                      value:
                        stringValue: $an-envvar-value-$$-suffix $$
                  startTimeUnixNano: "1000000"
                  timeUnixNano: "2000000"
                - asInt: "73723904"
                  attributes:
                    - key: decuple-dollar-curly-braces
                      value:
                        stringValue: $$$$${AN_ENVVAR}-$$$$$-suffix $$$$$
                    - key: double-dollar-curly-braces
                      value:
                        stringValue: ${AN_ENVVAR}-$-suffix $
                    - key: nonuple-dollar-curly-braces
                      value:
                        stringValue: $$$$an-envvar-value-$$$$$-suffix $$$$$
                    - key: octuple-dollar-curly-braces
                      value:
                        stringValue: $$$${AN_ENVVAR}-$$$$-suffix $$$$
                    - key: quadruple-dollar-curly-braces
                      value:
                        stringValue: $${AN_ENVVAR}-$$-suffix $$
                    - key: quintuple-dollar-curly-braces
                      value:
                        stringValue: $$an-envvar-value-$$$-suffix $$$
                    - key: septuple-dollar-curly-braces
                      value:
                        stringValue: $$$an-envvar-value-$$$$-suffix $$$$
                    - key: sextuple-dollar-curly-braces
                      value:
                        stringValue: $$${AN_ENVVAR}-$$$-suffix $$$
                    - key: single-dollar-curly-braces
                      value:
                        stringValue: an-envvar-value-$-suffix $
                    - key: state
                      value:
                        stringValue: slab_unreclaimable
                    - key: triple-dollar-curly-braces
                      value:
                        stringValue: $an-envvar-value-$$-suffix $$
                  startTimeUnixNano: "1000000"
                  timeUnixNano: "2000000"
                - asInt: "4796518400"
                  attributes:
                    - key: decuple-dollar-curly-braces
                      value:
                        stringValue: $$$$${AN_ENVVAR}-$$$$$-suffix $$$$$
                    - key: double-dollar-curly-braces
                      value:
                        stringValue: ${AN_ENVVAR}-$-suffix $
                    - key: nonuple-dollar-curly-braces
                      value:
                        stringValue: $$$$an-envvar-value-$$$$$-suffix $$$$$
                    - key: octuple-dollar-curly-braces
                      value:
                        stringValue: $$$${AN_ENVVAR}-$$$$-suffix $$$$
                    - key: quadruple-dollar-curly-braces
                      value:
                        stringValue: $${AN_ENVVAR}-$$-suffix $$
                    - key: quintuple-dollar-curly-braces
                      value:
                        stringValue: $$an-envvar-value-$$$-suffix $$$
                    - key: septuple-dollar-curly-braces
                      value:
                        stringValue: $$$an-envvar-value-$$$$-suffix $$$$
                    - key: sextuple-dollar-curly-braces
                      value:
                        stringValue: $$${AN_ENVVAR}-$$$-suffix $$$
                    - key: single-dollar-curly-braces
                      value:
                        stringValue: an-envvar-value-$-suffix $
                    - key: state
                      value:
                        stringValue: used
                    - key: triple-dollar-curly-braces
                      value:
                        stringValue: $an-envvar-value-$$-suffix $$
                  startTimeUnixNano: "1000000"
                  timeUnixNano: "2000000"
            unit: By
        scope:
          name: github.com/open-telemetry/opentelemetry-collector-contrib/receiver/hostmetricsreceiver/internal/scraper/memoryscraper
          version: v0.97.0
